### PR TITLE
Cleanup test_pthread_cleanup.cpp can convert to btest_exit. NFC

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

File: tests/pthread/test_pthread_cleanup.cpp [hunk 1]
```diff
@@ -16,95 +16,83 @@
...
+  s = pthread_create(&thr[2], NULL, thread_start2, (void*)1);
+  assert(s == 0);
+  pthread_join(thr[2], 0);
 // TODO
 //   s = pthread_create(&thr[3], NULL, thread_start3, (void*)1);
 //   assert(s == 0);
 //   s = pthread_cancel(thr[3]);
 //   assert(s == 0);
-   pthread_cleanup_pop(1);
-   EM_ASM(console.log('Cleanup state variable: ' + $0), cleanupState);
```
Comment (aheejin) [inline tests/pthread/test_pthread_cleanup.cpp:101]: This is not necessary anymore?

**Review by aheejin — COMMENTED** [review]
Can we possibly clang-format all c/cpp test files in a PR? We often end up mixing formatting changes with functional changes in a PR..

File: tests/pthread/test_pthread_cleanup.cpp (hunk 1 above)
Comment (sbc100) [reply tests/pthread/test_pthread_cleanup.cpp:101]: You mean the use of EM_ASM over printf?  I  don't think here is any reason not to just use printf here, is simpler, and we already use printf elsewhere in this test.

**Comment by sbc100** [comment]
> Can we possibly clang-format all c/cpp test files in a PR? We often end up mixing formatting changes with functional changes in a PR..

I thought about it, and I ever created a few experimental PRs to do that, at least for all the pthread tests.  But I always ran into one road block or another.  Normally a case where clang format seemed to make this things worse (for example EM_ASM I think can confuse it).

For now, I decided to take a more incremental approach.   I would still like to do an auto-format of the tests one day...

**Comment by sbc100** [comment]
For example, I decided to clean up this test because I happen to be working on pthread cleanup code in others changes right now.

**Review by aheejin — APPROVED** [review]

**Comment by kripken** [comment]
One of the posix tests failed when this landed, and the same failure happened on the next landing after it.

https://app.circleci.com/pipelines/github/emscripten-core/emscripten/14112/workflows/9d718bdc-260e-4a41-a1bc-eadb3e0d37dd/jobs/436542
https://app.circleci.com/pipelines/github/emscripten-core/emscripten/14115/workflows/2d2e6002-94a5-4f64-8fb4-e611c5596f89/jobs/436603

However, it seems like this PR can't cause that, which is odd... so might be a coincidence?

**Comment by sbc100** [comment]
Yes, it can't be this change that caused that.

**Comment by kleisauke** [comment]
I opened #14508 to keep track of this flaky test. PR #14509 disables this test for now.

**Pull request merged**